src/UI/FrontEnd.java

```diff
@@ -100,47 +109,61 @@ private void initializeWebAccountViewSection(JPanel websiteAccounts) {
 		JLabel websiteAccountSearchHeader = new JLabel("Search Account Information");
 		websiteAccountSearchHeader.setBounds(89, 43, 175, 16);
 		websiteAccounts.add(websiteAccountSearchHeader);
-		
+
 		websiteAccountSearchNicknameInput = new JTextField();
 		websiteAccountSearchNicknameInput.setBounds(112, 71, 130, 26);
 		websiteAccounts.add(websiteAccountSearchNicknameInput);
 		websiteAccountSearchNicknameInput.setColumns(10);
-		
-		JButton websiteAccountSearchButton = new JButton("Search");
+
+		websiteAccountSearchButton = new JButton("Search");
+
 		websiteAccountSearchButton.setBounds(248, 71, 73, 29);
 		websiteAccounts.add(websiteAccountSearchButton);
-		
+
 		JLabel websiteAccountSearchLabel = new JLabel("Nickname:");
 		websiteAccountSearchLabel.setFont(new Font("Lucida Grande", Font.PLAIN, 12));
 		websiteAccountSearchLabel.setBounds(47, 76, 67, 16);
 		websiteAccounts.add(websiteAccountSearchLabel);
-		
+
 		JTextArea websiteAccountNicknames = new JTextArea();
 		websiteAccountNicknames.setText("will hold account \nnicknames in \nnext iteration");
 		websiteAccountNicknames.setBounds(47, 109, 130, 257);
 		websiteAccounts.add(websiteAccountNicknames);
-		
+
 		JLabel websiteAccountViewUsernameLabel = new JLabel("Username:");
 		websiteAccountViewUsernameLabel.setFont(new Font("Lucida Grande", Font.PLAIN, 12));
 		websiteAccountViewUsernameLabel.setBounds(189, 181, 75, 16);
 		websiteAccounts.add(websiteAccountViewUsernameLabel);
-		
+
 		JLabel websiteAccountViewPasswordLabel = new JLabel("Password:");
 		websiteAccountViewPasswordLabel.setFont(new Font("Lucida Grande", Font.PLAIN, 12));
 		websiteAccountViewPasswordLabel.setBounds(189, 209, 61, 16);
 		websiteAccounts.add(websiteAccountViewPasswordLabel);
-		
+
 		//Label for displaying the username when it is gotten
-		JLabel websiteAccountViewUsernameResult = new JLabel("");
-		websiteAccountViewUsernameResult.setBounds(259, 181, 61, 16);
+		websiteAccountViewUsernameResult = new JLabel("");
+		websiteAccountViewUsernameResult.setBounds(259, 181, 155, 16);
 		websiteAccounts.add(websiteAccountViewUsernameResult);
-		
+
 		//label for displaying the password when it is gotten
-		JLabel websiteAccountViewPasswordResult = new JLabel("");
-		websiteAccountViewPasswordResult.setBounds(260, 209, 61, 16);
+		websiteAccountViewPasswordResult = new JLabel("");
+		websiteAccountViewPasswordResult.setBounds(260, 209, 154, 16);
 		websiteAccounts.add(websiteAccountViewPasswordResult);
+
+		websiteAccountSearchButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				WebsiteAccount website = UIController.getWebsiteAccountInfo(websiteAccountSearchNicknameInput.getText());
+				if(website != null) {
+					websiteAccountViewUsernameResult.setText(website.getWebsiteLogin());
+					websiteAccountViewPasswordResult.setText(website.getWebsitePassword());
+				}else {
+					websiteAccountViewUsernameResult.setText("Site doesn't exist");
```

Comment:
should this be JOptionPane.showMessageDialog(frame, "You have no website stored under " + websiteAccountSearchNicknameInput.getText());

and then probably clear the search after the if and else? websiteAccountSearchNicknameInput.setText("");

Reply:
probably